mesonbuild/interpreter/interpreter.py

```diff
@@ -1149,11 +1148,27 @@ def func_project(self, node: mparser.FunctionNode, args: T.Tuple[str, T.List[str
                 raise InterpreterException(f'Meson version is {cv} but project requires {pv}')
             mesonlib.project_meson_versions[self.subproject] = kwargs['meson_version']
 
-        if os.path.exists(self.option_file):
+        # Load "meson.options" before "meson_options.txt", and produce a warning if
+        # it is being used with an old version. I have added check that if both
+        # exist the warning isn't raised
+        option_file = os.path.join(self.source_root, self.subdir, 'meson.options')
+        old_option_file = os.path.join(self.source_root, self.subdir, 'meson_options.txt')
+
+        if os.path.exists(option_file):
+            if os.path.exists(old_option_file):
+                if os.path.samefile(option_file, old_option_file):
+                    mlog.debug("Not warning about meson.options with version minimum < 1.1 because meson_options.txt also exists")
+                else:
+                    mlog.error("meson.options and meson_options.txt both exist, but are not the same file.")
```

Comment:
I wonder if we need the debug path at all now.

As for the error, this seems wrong, because mlog.error doesn't intrinsically error out -- it simply prints the word ERROR, and triggers --fatal-meson-warnings. So, this would usually not actually error out. Shouldn't we just raise MesonException (or suitable subclass) here?

Reply:
Hmm, you're right. I thought `mlog.error` always raised, but it doesn't. I wonder if we should change that tbh. `mlog.error` seems like it shouldn't allow you to opt out, while `mlog.warning` does...